### PR TITLE
Fix GAMESS parser when there are >100 scf iterations

### PR DESCRIPTION
**Pull request by simonaxelrod**
Hi there,
I noticed a bug when there are more than 100 scf iterations in GAMESS. Please let me know how my fix looks.
Thanks,
Simon

### CODE REVIEW

File: cclib/parser/gamessparser.py [hunk 1]
```diff
@@ -689,14 +689,26 @@ def extract(self, inputfile, line):
                 except ValueError:
                     pass
                 else:
-                    values.append([float(line.split()[self.scf_valcol])])
+                    # if there were 100 iterations or more, the first part of the line
+                    # will look like 10099, 101100, 102101, etc., with no spaces between
+                    # the numbers. We can check if this is the case by seeing if the first
+                    # number on the line exceeds 10000.
+                    gt_100_its = int(line.split()[0]) > 10000
```
Comment (berquist) [inline cclib/parser/gamessparser.py:696]: Can you store the result of `line.split()` so it can be used here and reused again on lines 698 and 702?

**Review by berquist — CHANGES_REQUESTED** [review]
Hello Simon, thanks for the contribution! I don't see any obvious problems, though some of the naming is confusing. If you could submit an example logfile that has this problem to https://github.com/cclib/cclib-data/ via a PR, we can give some naming suggestions.

File: cclib/parser/gamessparser.py (hunk 1 above)
Comment (simonaxelrod) [reply cclib/parser/gamessparser.py]: Done!

**Comment by simonaxelrod** [comment]
Made the changes, and also made a PR request for cclib-data here https://github.com/cclib/cclib-data/pull/147. Let me know how this looks

**Review by berquist — CHANGES_REQUESTED** [review]
Having seen the output in https://github.com/cclib/cclib-data/pull/147, this makes more sense to me now. I was going to say that this could be avoided with using explicit indexing, but because of the need for `scf_valcol`, I can't see that solution being any shorter or easier to understand. So this looks good.

I've confirmed that the regression didn't parse before this patch and now it does. Could you do one more thing and add a test to https://github.com/cclib/cclib/blob/master/test/regression.py for the file you've added to cclib-data? Something like a check for how many SCF iterations it has is fine.

**Comment by simonaxelrod** [comment]
Great, thank you! I've also added a regression test which checks the number of scf iterations

**Review by berquist — APPROVED** [review]
Excellent! The regression test looks good, I've tested that it works. I'm going to hold off merging until we decide on what to do about our free Travis CI credits that have run out...

File: cclib/parser/gamessparser.py [hunk 2]
```diff
@@ -689,14 +689,26 @@ def extract(self, inputfile, line):
...
+                    # the numbers. We can check if this is the case by seeing if the first
+                    # number on the line exceeds 10000.
+                    base_split = line.split()
+                    gt_100_its = int(base_split[0]) > 10000
+                    if gt_100_its:
+                        first_two = [base_split[0][:3], base_split[0][3:]]
+                        split = [*first_two, *base_split[1:]]
+                    else:
+                        split = base_split
+                    values.append([float(split[self.scf_valcol])])
```
[the remark this answers is not in the document]
Comment (shivupa) [reply cclib/parser/gamessparser.py:703]: THis would get rid of the if statement

**Comment by shivupa** [comment]
Sorry we implemented this because it was holding up some CI testing things. Thanks for submitting the PR!

**Comment by simonaxelrod** [comment]
No problem, thanks for finishing it up!

**Pull request merged**